### PR TITLE
Misc 5

### PR DESCRIPTION
**Pull request by jschueller**
- #2525 
- #2552: fix binomial quantile
- #2541:

Régis: "At this point you can add a 1st order correction. You solved $F(x_p)=p-\epsilon$, and you are looking for $\eta$ such that $F(x_p+\eta)=p$. But, up to the first order, $F(x_p+\eta)=F(x_p)+\eta F'(x_p)$, so $p=p-\epsilon+\eta F'(x_p)$, resulting into $\eta=\epsilon/F'(x_p)$. It works only if $F'(x_p)\neq 0$, which is the case for a continuous mixture but not for a discrete distribution or a mix of discrete and continuous distributions."
- replace poissinv code by boost.math (see #2557)
- split some keepIntact methods (#2218)
- fix EGO stopping criterion
- #2571
- fix inversewishart variance
- #2524 (& NormalGamma)

### CODE REVIEW

File: lib/src/Uncertainty/Distribution/BayesDistribution.cxx [hunk 1]
```diff
@@ -403,29 +400,28 @@ void BayesDistribution::computeCovariance() const
...
   else integrator = GaussLegendre(Indices(2, static_cast<UnsignedInteger>(std::ceil(std::sqrt(1.0 * integrationNodesNumber_)))));
+
+  // The conditioning covariance can be reused as is
+  const UnsignedInteger conditioningDimension = conditioningDistribution_.getDimension();
+  const CovarianceMatrix conditioningCovariance(conditioningDistribution_.getCovariance());
+  for (UnsignedInteger i = 0; i < conditioningDimension; ++ i)
+    for (UnsignedInteger j = 0; j <= i; ++ j)
+      covariance_(i, j) = conditioningCovariance(i, j);
+
   // Integrate over the interval (-inf, y] of the conditioning Gamma distribution
```
Comment (regislebrun) [inline lib/src/Uncertainty/Distribution/BayesDistribution.cxx:416]: Strange comment...

**Review by regislebrun — APPROVED** [review]
Good job. I checked only the C++ part, good to go for me.

File: lib/src/Uncertainty/Distribution/BayesDistribution.cxx (hunk 1 above)
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/BayesDistribution.cxx:416]: I didnt get either

**Comment by jschueller** [comment]
hi @sofianehaddad can we merge this one ?

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**